crates/relayer/src/link/relay_path.rs

```diff
@@ -1381,11 +1389,31 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> RelayPath<ChainA, ChainB> {
         dst_info: &ChainStatus,
         height: Height,
     ) -> Result<(Option<Any>, Option<Any>), LinkError> {
+        crate::time!(
+            "build_recv_or_timeout_from_send_packet_event",
+            {
+                "src_channel": event.packet.source_channel,
+                "dst_channel": event.packet.destination_channel,
+            }
+        );
         let timeout = self.build_timeout_from_send_packet_event(event, dst_info)?;
         if timeout.is_some() {
             Ok((None, timeout))
         } else {
-            Ok((self.build_recv_packet(&event.packet, height)?, None))
+            match event
+                .packet
+                .validate_fields(self.max_memo_size, self.max_receiver_size)
+            {
+                Ok((0, 0)) => Ok((self.build_recv_packet(&event.packet, height)?, None)),
+                Ok((memo_size, receiver_size)) => {
+                    warn!("memo and/or receiver fields were bigger than maximum allowed. Memo field {memo_size}, maximum allowed {}. Receiver field {receiver_size}, maximum allowed {}", self.max_memo_size, self.max_receiver_size);
+                    Ok((None, None))
+                }
+                Err(e) => {
+                    warn!("error validating fields, will simply relay message: {e}");
+                    Ok((self.build_recv_packet(&event.packet, height)?, None))
```

Comment:
Will this warning show for all non-ICS-20 packets? That would be a bit noisy.

We should only show a warning/error if we see that we have an ICS-20 packet and:
- are unable to decode it
- the memo or the receiver fields are too large

What do you think?